engine/launcher/engine_server_launcher/engine_server_launcher.go

```diff
@@ -118,15 +121,22 @@ func (launcher *EngineServerLauncher) LaunchWithCustomVersion(
 		return nil, nil, stacktrace.Propagate(err, "An error occurred generating the engine server's environment variables")
 	}
 
+	shouldStartInDebugMode := automaticallyDetectIfShouldRunInDebugMode(imageVersionTag)
+
 	engine, err := launcher.kurtosisBackend.CreateEngine(
 		ctx,
 		containerImage,
 		imageVersionTag,
 		grpcListenPortNum,
 		envVars,
+		shouldStartInDebugMode,
 	)
 	if err != nil {
 		return nil, nil, stacktrace.Propagate(err, "An error occurred launching the engine server container with environment variables '%+v'", envVars)
 	}
 	return engine.GetPublicIPAddress(), engine.GetPublicGRPCPort(), nil
 }
+
+func automaticallyDetectIfShouldRunInDebugMode(engineImageVersionTag string) bool {
```

Comment:
Is the image tag the best way to the debug flag here?

Reply:
It was the fastest way :), but good review I'm adding a better approach by adding and argument to define this.